torch/_subclasses/fake_tensor.py

```diff
@@ -166,8 +166,7 @@ def torch_decomp_decompositions(func):
     from torch._decomp import decomposition_table
 
     decompositions = torch._decomp.decompositions
-    decomp_attrs = [getattr(decompositions, attr) for attr in dir(decompositions)]
-    return decomposition_table[func] in decomp_attrs
+    return decomposition_table[func].__name__ in dir(decompositions)
```

Comment:
What's the issue here?

Reply:
Previously `register_decomposition` was changing the reference functions to allow for differences in `out` handling in torch public API vs aten API. We fixed the issue by only changing the function in the decomposition table and keeping the original function where it was defined in `torch/_decomp/decompositions.py`.

Since the function in the decomposition table is different to the function that is referred in `torch._decomp.decompositions`, this function thinks they are different. This change fixes that.

Reply:
Makes sense, can we do something more like `decomposition_table[func].__module__.startswith("torch._decomp")`? This would prevent picking up functions with the same name but defined in a different place.